readme.txt

This project aims to provide belarusian language support for cuneiform OCR system. 
Below is original readme of the cuneiform project.
----------------------------------------------------------------------------------
Cuneiform for Linux


Cuneiform is an multi-language OCR system originally developed
and open sourced by Cognitive Technologies. Cuneiform was
originally a Windows program, which was ported to Linux
by Jussi Pakkanen. 

This version of Cuneiform has been tested to work on the following
platforms.

Linux
FreeBSD
OS X
Windows using MSVC, MinGW, and Cygwin

The following people have sent patches or have otherwise helped the
project. If someone is missing, please let me know, so I can add them.

Keith Beaumont
Vincent Wagelaar
zanin
Alexander Schlegel
Alex Samorukov
yaleks
Serj Poltavskiy
René Rebe
Aleks Kuzemko
Tonal
Mike Ladwig
Dmitri Polevoy
Steven Van Ingelgem
Sven Eckelmann
Benjamin Kluck
raff
Julien
John A
Frik


Caveats

There are known limitations in this port. Among these are the following:

- it only works on x86 and amd64 processors
- there is no table recognition, because of
  https://bugs.launchpad.net/cuneiform-linux/+bug/260327

Patches to fix any of these issues are gladly accepted.


Compiling on unix

Extract the source and go to the root folder (the one this file is in).
Then type the following commands:

mkdir builddir
cd builddir
cmake -DCMAKE_BUILD_TYPE=debug ..
make
make install 

By default Cuneiform installs to /usr/local. You can specify a different prefix
by giving a command line switch "-DCMAKE_INSTALL_PREFIX=/what/ever/you/want"
to CMake.

Note that this does not use any optimizations. To enable them, replace
-DCMAKE_BUILD_TYPE=debug with -DCMAKE_BUILD_TYPE=release or with
-DCMAKE_BUILD_TYPE=relwithdebinfo. The latter builds with both optimization
and debug flags.


Compiling on OSX

Follow the instructions for unix above. You can also try the Xcode project
generator. It works but is not maintained.


Compiling on Windows

Run CMake. Point it to the directory you extracted the source to. 
Select a different directory for your build tree, You can not build inside
your source tree.

Select "Visual studio [the version you are using] project files", 
"MinGW makefiles", or "Cygwin makefiles" depending on your environment.
If you are using MinGW or Cygwin, set CMAKE_BUILD_TYPE to what you want
(probaly "debug" as above). Click "config", then "generate".

For Visual Studio, open the generated project files.
Then select "Build solution (F7)".

For MinGW and Cygwin, cd into your build tree and run "mingw32-make" or "make". 


Further info on configuration and running

If you have ImageMagick++ on your system, Cuneiform autodetects and builds
against it. Then Cuneiform can process any image that ImageMagick knows how
to open. Otherwise it can only read uncompressed BMP images.

If you want to run Cuneiform without installing it on your system, you
have to point the CF_DATADIR environment variable to a directory
containing the .dat files. These can be found in the "datafiles"
directory of the source package.

WINDOWS NOTE: Cuneiform tries to access its data files in the
directory that contains the DLLs. You can place Cuneiform
anywhere you want on your hard drive, just put the .dat files
in the same directory. This works also when using Cuneiform
as a library.


Running

After install you simply run.

cuneiform [-l language -o result_file -f [outputformat] extra_options ] <image_file>

Optional arguments are the following.

--dotmatrix uses a recognition mode optimised for text printed with a
dot matrix printer.

--fax uses a recognition mode optimised for text that has been faxed.

--singlecolumn disables page layout analysis and assumes that your
image consists of only one column of text.

If you do not define an output file with the -o switch, Cuneiform
writes the result to a file "cuneiform-out.[format]". The file extension
depends on your output format.

By default Cuneiform recognizes English text. To change the language use the
command line switch -l followed by your language string. To get a list of
supported languages type "cuneiform -l".

By default Cuneiform outputs plain text. There are several other output formats.
To get a list run the command "cuneiform -f".


Contact information

Project home page: https://launchpad.net/cuneiform-linux
Mailing list: https://launchpad.net/~cuneiform


debian/changelog
cuneiform (1.0.0+dfsg-2) unstable; urgency=low

  * Upload to unstable.
  * Explicitly build-depend on pkg-config. Thanks to Stefano Rivera for the
    bug report.
  * Add Vcs-* fields.
  * Use the standard C assert() macro, rather than custom Cuneiform one.
    [c-assert.diff]
  * Pass CFLAGS, CXXFLAGS and LDFLAGS (get from dpkg-buildflags) to cmake
    (closes: #608345). Thanks to Sami Liedes for the bug report.
    + Build depend on dpkg-dev (>= 1.15.7).
  * Pass --parallel to dh.
    + Bump debhelper minimum version to 7.4.10.
  * Update debian/copyright to the latest DEP-5 version.
  * Bump year in debian/copyright.
  * Explicitly link to GraphicsMagick (rather than via the ImageMagick
    compatibility layer).
  * Don't ship /usr/lib/cuneiform/*.so symlinks. These libraries are
    considered private, at least until #598616 is fixed.
  * Rename some private variables in debian/rules to make them lowercase.
  * Update patch headers.
  * Provide proper ‘build-arch’ and ‘build-indep’ targets in debian/rules.
  * Document input format in the manual page (closes: #572061). Thanks to
    Janusz S. Bień for the bug report.
  * Use ‘slv’ (rather than ‘slo’) as language code for Slovenian.
    [slovenian-slv.diff]
  * Fix package description: Slovenian is supported, Slovak is not.
  * Improve documentation of the language (-l) option (closes: #602512).
    Thanks to Jari Aalto for the bug report.
  * Install reasons file for vrms.

 -- Jakub Wilk <jwilk@debian.org>  Wed, 26 Jan 2011 21:53:07 +0100

cuneiform (1.0.0+dfsg-1) experimental; urgency=low

  * New upstream release (closes: #575419).
  * New maintainer (closes: #543893).
  * Document in README.source how to repackage upstream tarball.
  * Update debian/copyright.
    + Document that the package is auto-buildable.
    + Document which files were stripped from the upstream tarball.
    + Convert to the DEP-5 format.
  * Drop README.Debian, no longer needed.
  * Rewrite debian/rules from scratch using dh.
    + Bump debhelper minimum version to 7.4.4 (for cmake support).
    + Turn on MAKE_VERBOSE_MAKEFILE.
    + Disable call to dh_makeshlibs to avoid creation of spurious
      calls to ldconfig in postrm/postinst.
    + Add get-orig-source target.
  * Remove unused overrides.
  * Link to GraphicsMagick rather than ImageMagick:
    + Build-depend on graphicsmagick-libmagick-dev-compat.
    + Add patch to properly initialize the library. [graphicsmagick.diff]    
  * Bump standards version to 3.9.1 (no changes needed).
  * Force upgrade of ocrodjvu to at least 0.4.2; earlier versions of ocrodjvu
    don't support this version of Cuneiform.
  * List all the supported languages in the package description.
  * Update the manual page.
    + Document --dotmatrix, --fax and --singlecolumn options.
    + Document all output formats.
    + Don't use hyphen as minus sign.
    + Fix a few formatting issues.
  * Drop Vcs-* fields.
  * Add watch file.
  * Add patch from Dmitrijs Ledkovs to link the lao library with libm.
    [libm.diff]
  * Add patch for a for a more portable function to swap byte order.
    [swapbytes.diff]
  * Set Architecture to any; there's no good reason to limit it.
  * Run rudimentary tests at build time.

 -- Jakub Wilk <jwilk@debian.org>  Tue, 14 Sep 2010 15:53:54 +0200

cuneiform (0.7.0+dfsg.1-1) unstable; urgency=low

  * QA upload
  * Repacked upstream tarball without cuneiform_src/Kern/rout/wtypes.h
    since its licence has been doubted, and since it isn't used anyway.
    (closes: #583603). Removed mention of that file from debian/copyright.
  * Recompiled against libmagick++3 (closes: #575420).
  * Bumped dependency on debhelper to >= 7.0.50~.
  * Source format 3.0 (quilt): no change

 -- Ralf Treinen <treinen@debian.org>  Tue, 29 Jun 2010 20:12:53 +0200

cuneiform (0.7.0+dfsg-5) unstable; urgency=low

  * Updating package to standards version 3.8.3.
  * Removing vcs fields.
  * Orphaning package.

 -- Daniel Baumann <daniel@debian.org>  Thu, 27 Aug 2009 10:42:31 +0200

cuneiform (0.7.0+dfsg-4) unstable; urgency=low

  * Minimalizing rules file.

 -- Daniel Baumann <daniel@debian.org>  Wed, 29 Jul 2009 21:44:01 +0200

cuneiform (0.7.0+dfsg-3) unstable; urgency=low

  * Avoiding circular depends (Closes: #531707).
  * Updating standards version to 3.8.2.
  * Adding README.Debian with a note about non-free.

 -- Daniel Baumann <daniel@debian.org>  Fri, 10 Jul 2009 17:58:10 +0200

cuneiform (0.7.0+dfsg-2) unstable; urgency=low

  * Adding Autobuild field in control (Closes: #531363).
  * Updating version and date in manpage.
  * Building binaries with optimization.

 -- Daniel Baumann <daniel@debian.org>  Mon, 01 Jun 2009 09:18:01 +0200

cuneiform (0.7.0+dfsg-1) unstable; urgency=low

  * Using correct rfc-2822 date formats in changelog.
  * Updating to standards 3.8.1.
  * Merging upstream version 0.7.0+dfsg.
  * Passing libdir to cmake (Closes: #531277).

 -- Daniel Baumann <daniel@debian.org>  Sun, 31 May 2009 19:54:52 +0200

cuneiform (0.6.0+dfsg-1) unstable; urgency=low

  * Initial release (Closes: #496264).
  * Rebuild upstream tarball without win32 binaries.
  * The source for the .dat files is unknown, therefore uploading to non-free,
    see comments in http://bugs.debian.org/496264 from Jussi Pakkanen.

 -- Daniel Baumann <daniel@debian.org>  Mon, 06 Apr 2009 02:38:00 +0200


cuneiform_src/Kern/hrk/lang_def.h
/*
Copyright (c) 1993-2008, Cognitive Technologies
All rights reserved.

Разрешается повторное распространение и использование как в виде исходного кода,
так и в двоичной форме, с изменениями или без, при соблюдении следующих условий:

      * При повторном распространении исходного кода должны оставаться указанное
        выше уведомление об авторском праве, этот список условий и последующий
        отказ от гарантий.
      * При повторном распространении двоичного кода в документации и/или в
        других материалах, поставляемых при распространении, должны сохраняться
        указанная выше информация об авторском праве, этот список условий и
        последующий отказ от гарантий.
      * Ни название Cognitive Technologies, ни имена ее сотрудников не могут
        быть использованы в качестве средства поддержки и/или продвижения
        продуктов, основанных на этом ПО, без предварительного письменного
        разрешения.

ЭТА ПРОГРАММА ПРЕДОСТАВЛЕНА ВЛАДЕЛЬЦАМИ АВТОРСКИХ ПРАВ И/ИЛИ ДРУГИМИ ЛИЦАМИ "КАК
ОНА ЕСТЬ" БЕЗ КАКОГО-ЛИБО ВИДА ГАРАНТИЙ, ВЫРАЖЕННЫХ ЯВНО ИЛИ ПОДРАЗУМЕВАЕМЫХ,
ВКЛЮЧАЯ ГАРАНТИИ КОММЕРЧЕСКОЙ ЦЕННОСТИ И ПРИГОДНОСТИ ДЛЯ КОНКРЕТНОЙ ЦЕЛИ, НО НЕ
ОГРАНИЧИВАЯСЬ ИМИ. НИ ВЛАДЕЛЕЦ АВТОРСКИХ ПРАВ И НИ ОДНО ДРУГОЕ ЛИЦО, КОТОРОЕ
МОЖЕТ ИЗМЕНЯТЬ И/ИЛИ ПОВТОРНО РАСПРОСТРАНЯТЬ ПРОГРАММУ, НИ В КОЕМ СЛУЧАЕ НЕ
НЕСЁТ ОТВЕТСТВЕННОСТИ, ВКЛЮЧАЯ ЛЮБЫЕ ОБЩИЕ, СЛУЧАЙНЫЕ, СПЕЦИАЛЬНЫЕ ИЛИ
ПОСЛЕДОВАВШИЕ УБЫТКИ, СВЯЗАННЫЕ С ИСПОЛЬЗОВАНИЕМ ИЛИ ПОНЕСЕННЫЕ ВСЛЕДСТВИЕ
НЕВОЗМОЖНОСТИ ИСПОЛЬЗОВАНИЯ ПРОГРАММЫ (ВКЛЮЧАЯ ПОТЕРИ ДАННЫХ, ИЛИ ДАННЫЕ,
СТАВШИЕ НЕГОДНЫМИ, ИЛИ УБЫТКИ И/ИЛИ ПОТЕРИ ДОХОДОВ, ПОНЕСЕННЫЕ ИЗ-ЗА ДЕЙСТВИЙ
ТРЕТЬИХ ЛИЦ И/ИЛИ ОТКАЗА ПРОГРАММЫ РАБОТАТЬ СОВМЕСТНО С ДРУГИМИ ПРОГРАММАМИ,
НО НЕ ОГРАНИЧИВАЯСЬ ЭТИМИ СЛУЧАЯМИ), НО НЕ ОГРАНИЧИВАЯСЬ ИМИ, ДАЖЕ ЕСЛИ ТАКОЙ
ВЛАДЕЛЕЦ ИЛИ ДРУГОЕ ЛИЦО БЫЛИ ИЗВЕЩЕНЫ О ВОЗМОЖНОСТИ ТАКИХ УБЫТКОВ И ПОТЕРЬ.

Redistribution and use in source and binary forms, with or without modification,
are permitted provided that the following conditions are met:

    * Redistributions of source code must retain the above copyright notice,
      this list of conditions and the following disclaimer.
    * Redistributions in binary form must reproduce the above copyright notice,
      this list of conditions and the following disclaimer in the documentation
      and/or other materials provided with the distribution.
    * Neither the name of the Cognitive Technologies nor the names of its
      contributors may be used to endorse or promote products derived from this
      software without specific prior written permission.

THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE
FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
*/

#ifndef _LANGDEF_INCLUDED_
#define _LANGDEF_INCLUDED_

#if defined (__NT__) || defined (_WINNT_) || defined(__MINGW32__) // 06.09.00 E.P.
  #undef LANG_ENGLISH
  #undef LANG_GERMAN
  #undef LANG_FRENCH
  #undef LANG_RUSSIAN
  #undef LANG_SWEDISH
  #undef LANG_SPANISH
  #undef LANG_ITALIAN
  #undef LANG_UKRAINIAN
  #undef LANG_TOTAL
  #undef LANG_SERBIAN
  #undef LANG_CROATIAN
  #undef LANG_DANISH
  #undef LANG_PORTUGUESE
  #undef LANG_DUTCH
  #undef LANG_TOTAL
  #undef LANG_POLISH
  #undef LANG_UZBEK
  #undef LANG_KAZ
  #undef LANG_KAZ_ENG
  #undef LANG_CZECH
  #undef LANG_ROMAN
  #undef LANG_HUNGAR
  #undef LANG_BULGAR
  #undef LANG_SLOVENIAN
  #undef LANG_LATVIAN
  #undef LANG_LITHUANIAN
  #undef LANG_ESTONIAN
  #undef LANG_TURKISH
  #undef LANG_BELARUSIAN
#endif

enum Languages {
    LANG_ENGLISH    = 0,
    LANG_GERMAN     = 1,
    LANG_FRENCH     = 2,
    LANG_RUSSIAN    = 3,
    LANG_SWEDISH    = 4,
    LANG_SPANISH    = 5,
    LANG_ITALIAN    = 6,
    LANG_RUSENG     = 7,
    LANG_UKRAINIAN  = 8,
    LANG_SERBIAN    = 9,
    LANG_CROATIAN   = 10,
    LANG_POLISH     = 11,
    LANG_DANISH     = 12,
    LANG_PORTUGUESE = 13,
    LANG_DUTCH      = 14,
    LANG_DIG        = 15,
    LANG_UZBEK      = 16,
    LANG_KAZ        = 17,
    LANG_KAZ_ENG    = 18,
    LANG_CZECH      = 19,
    LANG_ROMAN      = 20,
    LANG_HUNGAR     = 21,
    LANG_BULGAR     = 22,
    LANG_SLOVENIAN  = 23,
    LANG_LATVIAN    = 24,
    LANG_LITHUANIAN = 25,
    LANG_ESTONIAN   = 26,
    LANG_TURKISH    = 27,
    LANG_BELARUSIAN = 28,
    LANG_TOTAL      = 29,
    LG_LAT          =  1,
    LG_CYR          =  2,
};

#endif


